actors/evm/src/lib.rs

```diff
@@ -61,13 +65,26 @@ pub enum Method {
 
 pub struct EvmContractActor;
 
+/// Returns a tombstone for the currently executing message.
+pub(crate) fn current_tombstone(rt: &impl Runtime) -> Tombstone {
+    Tombstone { origin: rt.message().origin().id().unwrap(), nonce: rt.message().nonce() }
+}
+
+/// Returns true if the contract is "dead". A contract is dead if:
+///
+/// 1. It has a tombstone.
+/// 2. It's tombstone is not from the current message execution.
```

Comment:
Could you elaborate on this for our future selves? The tombstoning state transition is tricky.